go/processor/floria/processor_test.go

```diff
@@ -113,3 +114,99 @@ func TestProcessor_BuyGasInsufficientBalance(t *testing.T) {
 		t.Errorf("buyGas did not fail with insufficient balance")
 	}
 }
+
+func TestGasUsed(t *testing.T) {
+	tests := []struct {
+		sender          tosca.Address
+		expectedGasUsed tosca.Gas
+	}{
+		{
+			sender:          tosca.Address{},
+			expectedGasUsed: 500,
+		},
+		{
+			sender:          tosca.Address{1},
+			expectedGasUsed: 550,
+		},
+		{
+			sender:          tosca.Address{42},
+			expectedGasUsed: 550,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("sender%v", test.sender), func(t *testing.T) {
+			transaction := tosca.Transaction{
+				Sender:   test.sender,
+				GasLimit: 1000,
+			}
+
+			gasLeft := tosca.Gas(500)
+			actualGasUsed := gasUsed(transaction, gasLeft)
+
+			if actualGasUsed != test.expectedGasUsed {
+				t.Errorf("gasUsed returned incorrect result, got: %d, want: %d", actualGasUsed, test.expectedGasUsed)
+			}
+		})
+	}
+}
+
+func TestProcessor_SetupGasBilling(t *testing.T) {
+	tests := map[string]struct {
+		recipient       *tosca.Address
+		input           []byte
```

Comment:
dont't we have a `tosca.Code` for this types ?

Reply:
This is the input and not code, we handle it as a byte slice everywhere.